dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,19 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val transitions: Set[Transition], val start: State, val accept: Set[State]) {
+  def accepts(input: String): Boolean = {
+    var currentState = start
+    for (symbol <- input) {
+      val transitionOpt = transitions.find(t => t.from == currentState && t.symbol == symbol)
+      transitionOpt match {
```

Comment:
I like the idea of checking if there is a transition option. I just always set currentState with to the equivalent of transition.to here and then I check if accept contains that but doing this makes sure that there is a transition, I'm not sure how contains(None) would behave so maybe it would have been better for me to avoid leaving room for that error in my code.